staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -672,3 +675,34 @@ func CopyConfig(config *Config) *Config {
 	}
 	return c
 }
+
+func DefaultCodecFactoryForScheme(scheme *runtime.Scheme) serializer.CodecFactory {
```

Comment:
The approach in this PR looks reasonable to me.  I haven't thought carefully about if this is the best package for this function, but it doesn't seem unreasonable.